app/transactions/ams/steps/create_aapb_admin_data.rb

```diff
@@ -51,22 +51,26 @@ def save_aapb_admin_data(change_set)
       end
 
       def set_admin_data_attributes(admin_data, change_set)
-        AdminData.attributes_for_update.each do |k|
-          # Some attributes are serialized on AdminData, so always send an array
-          k_string = k.to_s
-          if should_empty_admin_data_value?(k, change_set)
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array.new) : admin_data.send("#{k}=", nil)
-          elsif change_set.fields[k_string].present?
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array(change_set.fields[k_string])) : admin_data.send("#{k}=", change_set.fields[k_string].to_s)
+        AdminData.attributes_for_update.each do |field|
+          field = field.to_s
+          # Convert emtpy strings to nil, whether the value is a scalar or an array
+          new_admin_data_value = if change_set.fields[field].respond_to?(:select)
+            change_set.fields[field].reject {|v| v.blank? }
+          elsif change_set.fields[field].blank?
+            nil
+          end
+
+          if can_empty_field?(field) || new_admin_data_value.present?
+            admin_data.write_attribute(field, new_admin_data_value)
```

Comment:
Since we only write conditionally, I'd love to see an `else` which logs fields we choose _not_ to write. Historically, this has helped immensely when trying to track down issues along the lines of "I see my data going in but not coming out" 

```suggestion
          if can_empty_field?(field) || new_admin_data_value.present?
            admin_data.write_attribute(field, new_admin_data_value)
          else
            Rails.logger.debug("Chose not to update AdminData #{admin_data.id}'s field #{field} with value(s): #{change_set.fields[field]}")
```